modules/navier_stokes/src/actions/NSFVAction.C

```diff
@@ -0,0 +1,2259 @@
+                        "The value used for pinning the pressure (point value/domain average).");
+
+  params.addParam<bool>("boussinesq_approximation", false, "True to have Boussinesq approximation");
+
+  params.addRangeCheckedParam<Real>(
+      "ref_temperature",
+      273.15,
+      "ref_temperature > 0.0",
+      "Value for reference temperature in case of Boussinesq approximation");
+  params.addParam<MooseFunctorName>(
+      "thermal_expansion", NS::alpha, "The name of the thermal expansion");
+
+  params.addParamNamesToGroup("pinned_pressure_type pinned_pressure_point pinned_pressure_value "
+                              "ref_temperature boussinesq_approximation",
+                              "Momentum controls");
+
+  /**
+   * Equations used to set up the energy equation/enthalpy equation if it is required.
+   */
+
+  params.addParam<FunctionName>(
+      "initial_temperature", "300", "The initial temperature, assumed constant everywhere");
+
+  params.addParam<MooseFunctorName>(
+      "thermal_conductivity", NS::k, "The name of the fluid thermal conductivity");
+
+  params.addParam<MooseFunctorName>("specific_heat", NS::cp, "The name of the specific heat");
+
+  MultiMooseEnum en_inlet_types("fixed-temperature flux-mass flux-velocity heatflux", "heatflux");
+  params.addParam<MultiMooseEnum>("energy_inlet_types",
+                                  en_inlet_types,
+                                  "Types for the inlet boundaries for the energy equation.");
+
+  params.addParam<std::vector<std::string>>(
+      "energy_inlet_function",
+      std::vector<std::string>(),
+      "Functions for fixed-value boundaries in the energy equation.");
+
+  MultiMooseEnum en_wall_types("fixed-temperature heatflux symmetry", "symmetry");
+  params.addParam<MultiMooseEnum>(
+      "energy_wall_types", en_wall_types, "Types for the wall boundaries for the energy equation.");
+
+  params.addParam<std::vector<FunctionName>>(
+      "energy_wall_function",
+      std::vector<FunctionName>(),
+      "Functions for Dirichlet/Neumann boundaries in the energy equation.");
+
+  params.addParam<std::vector<SubdomainName>>(
+      "ambient_convection_blocks",
+      std::vector<SubdomainName>(),
+      "The blocks where the ambient convection is present.");
+
+  params.addParam<std::vector<MooseFunctorName>>(
+      "ambient_convection_alpha",
+      std::vector<MooseFunctorName>(),
+      "The heat exchange coefficients for each block in 'ambient_convection_blocks'.");
+
+  params.addParam<std::vector<MooseFunctorName>>(
+      "ambient_temperature",
+      std::vector<MooseFunctorName>(),
+      "The ambient temperature for each block in 'ambient_convection_blocks'.");
+
+  params.addParam<CoupledName>(
+      "external_heat_source",
+      "The name of a functor which contains the external heat source for the energy equation.");
+  params.addParam<Real>(
+      "external_heat_source_coeff", 1.0, "Multiplier for the coupled heat source term.");
```

Comment:
Added! Also, I renamed the parameter groups to avoid instances like this:
```
Material Property Names Parameters
Numerical Scheme Parameters Parameters
Boundary Condition Parameters Parameters
```

Reply:
didnt know it did that. I need to check my old PRs!